js/form.js

```diff
@@ -0,0 +1,25 @@
+//On the form page, display an alert that says "Thank you. The form information has been received" when the form has been submitted.
+const form = document.getElementById('dogdataForm')
+const formAlertSubmitt = document.getElementById('submitForm');
+formAlertSubmitt.setAttribute('type','submit');
+formAlertSubmitt.textContent = 'Submit';
+
+
+form.addEventListener('submit', function(e){
+  //On the form page, display an alert that says "Thank you. The form information has been received" when the form has been submitted.
+  event.preventDefault();
+  alert('Thank you. The form information has been received.');
+
+  //store data in an array or object
+  const checkoutFormData = form.elements;
+  console.log(checkoutFormData);
+  const submitData = {};
+
+  for(i = 0; i < checkoutFormData.length-1; i++){
+    const fieldData = form.element[i].value;
+    const fieldName = form[i].id;
```

Comment:
You had an excellent question on why this works. In JS, there are two ways we typically access properties and values. The first is by using a dot. So if I want the ID of something I can do `something.id`. When it comes to an array of something, we use index. So when you access form[1], that is accessing the 2nd element with the form's array of elements. The form itself is made up of various elements as we can see above when you printed the elements out. So form.element is not an array. Therefore you can't access an index of an array that doesn't exist. Checkout a JSFiddle I made demonstrating this: [https://jsfiddle.net/a5Lhgocs/1/](https://jsfiddle.net/a5Lhgocs/1/)